File: default/src/app/_models/coursecategory.ts
export interface CourseCategory {
    ID: number;
    Name: string;
    Alias: string;
    Description: string;
    ParentID?: number;
    DisplayOrder?: number;
    CreatedDate: Date;
    CreatedBy: string;
    UpdatedDate?: Date;
    UpdatedBy: string;
    Status: boolean;
}

File: Learning.API/Models/CourseCategory.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Learning.API.Models.Abstract;

namespace Learning.API.Models
{
    [Table("CourseCategories")]
    public class CourseCategory: Auditable
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { set; get; }

        [Required]
        [MaxLength(256)]
        public string Name { set; get; }

        [Required]
        [MaxLength(256)]
        public string Alias { set; get; }

        [MaxLength(500)]
        public string Description { set; get; }

        public int? ParentID { set; get; }
        public int? DisplayOrder { set; get; }

        public ICollection<Course> Courses { get; set; }


}
}